resources/migration-guide-v2-v5.md

```diff
@@ -0,0 +1,402 @@
+    PurseIdentifier.fromAccountHash(
+      privateKey.publicKey.accountHash()
+    )
+  );
+} catch (e) {}
+```
+
+Explore the new PurseIdentifier class [here](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/rpc/request.ts#L542).
+
+---
+
+### 7. **`DeployUtil` to `Deploy`**
+
+The `DeployUtil` module in the previous versions of casper-js-sdk has been replaced with the `Deploy` class in version 5.x. Functionality for deploy creation `DeployUtil` become [Deploy](../src/types/Deploy.ts) class with static methods. [See how to create Deploy here](../README.md#creating-a-legacy-deploy-1)
+
+### Key Differences
+
+1. **Module Structure**:
+   - The old DeployUtil module had multiple static utility methods.
+   - The new Deploy class organizes these functionalities as methods on a single object.
+2. **Serialization and Deserialization**:
+   - The new module uses TypedJSON for serialization/deserialization, ensuring type safety.
+3. **Improved Method Organization**:
+   - Methods are grouped logically, providing a cleaner and more intuitive API.
+
+### Additional Notes
+
+- The new Deploy class provides a more type-safe and modular approach, reducing errors and improving code readability.
+- Refer to the [Deploy Class Documentation](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/types/Deploy.ts) for a complete list of available methods.
+- Ensure that you update your dependencies and verify compatibility with the 5.x SDK before migrating.
+
+---
+
+### 8. **`Contracts` Abstraction Removed**
+
+In the newer version of `casper-js-sdk`, we’ve removed the abstraction previously used for interacting with smart contracts. To provide greater flexibility and clarity, we have introduced two distinct classes: `ContractHash` and `ContractPackageHash`. These allow developers to directly install and call smart contracts with greater control. Below are two popular examples to help you transition smoothly:
+
+**Example 1: Installing a Smart Contract on the Casper Network.**
+
+This example demonstrates how to install a new smart contract using the new SDK structure. You can now directly work with `ExecutableDeployItem.newModuleBytes()` to include your `.wasm` file and deployment arguments.
+
+```tsx
+import {
+  PrivateKey,
+  PublicKey,
+  Args,
+  DeployHeader,
+  ExecutableDeployItem,
+  Deploy,
+  DEFAULT_DEPLOY_TTL,
+  StoredContractByHash,
+  ContractHash
+} from 'casper-js-sdk';
+
+// Install a smart contract on a Casper Network
+export const install = (
+  wasm: Uint8Array,
+  args: Args,
+  paymentAmount: string,
+  sender: PublicKey,
+  chainName: string,
+  signingKey: PrivateKey
+): Deploy => {
+  const deployHeader = DeployHeader.default();
+  deployHeader.account = sender;
+  deployHeader.chainName = chainName;
+
+  const payment = ExecutableDeployItem.standardPayment(paymentAmount);
+  const deploy = Deploy.makeDeploy(
+    deployHeader,
+    payment,
+    ExecutableDeployItem.newModuleBytes(wasm, args)
```

Comment:
I would create a session variable above and use it here for clarity. And to be closer to the next example.